### PR TITLE
NF - add tracking workflow parameters

### PR DESCRIPTION
**Pull request by gabknight**
- dipy_track_local
  - added step_size option
  - change default algorithm from deterministicMaxDG to EuDX (#1787)

- Update the DOC

this addresses #1758 .

### CODE REVIEW

**Comment by skoudoro** [comment]
LGTM, Thank you @gabknight. Do you plan to address the second part of #1766 (`Stopping_files...`) on this PR or on a new one?

**Comment by gabknight** [comment]
@skoudoro I will add the second part as well.

Do you have any idea what is the error for the last commit?

**Comment by gabknight** [comment]
I just rebased. The PR is ready to be reviewed/merged.

File: dipy/workflows/tracking.py [hunk 1]
```diff
@@ -88,18 +84,20 @@ def _core_run(self, stopping_path, stopping_thr, seeding_path,
...
+                                affine_to_rasmm=np.eye(4))
         save(tractogram, out_tract)
-
         logging.info('Saved {0}'.format(out_tract))
 
     def run(self, pam_files, stopping_files, seeding_files,
             stopping_thr=0.2,
             seed_density=1,
+            step_size=0.5,
             tracking_method="deterministic",
```
Comment (skoudoro) [inline dipy/workflows/tracking.py]: Can you replace "deterministic" by "eudx"

File: dipy/workflows/tracking.py [hunk 2]
```diff
@@ -23,8 +23,8 @@ class LocalFiberTrackingPAMFlow(Workflow):
     def get_short_name(cls):
         return 'track_local'
 
-    def _get_direction_getter(self, strategy_name, pam, pmf_threshold=0.1,
-                              max_angle=30.):
+    def _get_direction_getter(self, strategy_name, pam, pmf_threshold,
+                              max_angle):
```
Comment (skoudoro) [inline dipy/workflows/tracking.py]: it seems I forgot to add all parameters to the docstring. Can you add them (`pam`, `pmf_threshold`, `max_angle`)?

File: dipy/workflows/tracking.py [hunk 3]
```diff
@@ -115,17 +113,19 @@ def run(self, pam_files, stopping_files, seeding_files,
...
             wildcards to use multiple masks at once.
         stopping_files : string
-            Path of FA or other images used for stopping criteria for tracking.
+            Path to images (e.g. FA) used for stopping criteria for tracking.
         seeding_files : string
             A binary image showing where we need to seed for tracking.
         stopping_thr : float, optional
             Threshold applied to stopping volume's data to identify where
-             tracking has to stop (default 0.25).
+            tracking has to stop (default 0.25).
```
Comment (skoudoro) [inline dipy/workflows/tracking.py]: This space is on purpose because of the space stripping when you display the help

Comment (skoudoro) [reply dipy/workflows/tracking.py]: you can forget this comment

File: dipy/workflows/tracking.py (hunk 1 above)
Comment (gabknight) [reply dipy/workflows/tracking.py]: done. thx.

File: dipy/workflows/tracking.py (hunk 2 above)
Comment (gabknight) [reply dipy/workflows/tracking.py]: done.

**Comment by gabknight** [comment]
Thanks @skoudoro.

This addresses #1758 and #1787.

Let's keep the `Stopping_files` enhancement for another PR (#1811)

**Comment by skoudoro** [comment]
Thank you @gabknight

**Pull request merged**